### ISSUE TITLE
Search with POST: attributesToHighlight, attributesToCrop and attributesToRetrieve don't accept simple strings

### ISSUE THREAD

**Issue by eskombro** [meilisearch/MeiliSearch#866]
**Describe the bug**
As described in the [search parameters documentation](https://docs.meilisearch.com/guides/advanced_guides/search_parameters.html), the parameters `attributesToHighlight`, `attributesToCrop` and `attributesToRetrieve` can be either a simple string with `'*'`, or a string accepting coma separated attributes as in `"attribute1,attribute2,attribute3"`.

When using the POST route for search, MeilISearch is expecting an array of strings, and a simple string will throw an error from Serde deserialization, resulting in a 400 `bad_request` error

**To Reproduce**
1. Create an index called `new_index`
2. Index the following documents:

```
[
    {
        'id': 1,
        'name': 'Sam',
        'age': 20,
    },
    {
        'id': 2,
        'name': 'Max',
        'age': 40,
    },
    {
        'id': 3,
        'name': 'Tom',
        'age': 60,
    }
]
```
3. Do the following search request with a simple string:

```bash
curl -X POST --data '{ "q": "sam", "attributesToHighlight": "name" }'  http://localhost:7700/indexes/new_index/search
```

Response:
```
{"message":"Invalid JSON: invalid type: string \"name\", expected a sequence at line 1 column 45","errorCode":"bad_request","errorType":"invalid_request_error","errorLink":"https://docs.meilisearch.com/error/bad_request"}
```
4. Do the following search request with an array of string:

```bash
curl -X POST --data '{ "q": "sam", "attributesToHighlight": ["name"] }'  http://localhost:7700/indexes/new_index/search
```

Response:
```
{"hits":[{"id":1,"name":"Sam","age":20,"_formatted":{"id":1,"name":"<em>Sam</em>","age":20}}],"offset":0,"limit":20,"nbHits":1,"exhaustiveNbHits":false,"processingTimeMs":2,"query":"sam"}
```

As you can see in the example, `attributesToHighlight` needs to be an array for getting proper deserialization. 

**Expected behavior**
It should be possible to send a simple string

**Comment by curquiza** [comment]
After discussing with the team, this is normal, you can close the issue.

**Comment by MarinPostma** [comment]
@eskombro there are slight differences in the API of the POST and GET routes, the documentation for this new route will be available with the 0.13.0 :) (this pr https://github.com/meilisearch/documentation/pull/416)
